src/datepicker/datepicker.js

```diff
@@ -323,15 +332,7 @@ export default class Datepicker<T = Date> extends React.Component<DatepickerProp
     } else {
       const dateString = this.normalizeDashes(inputValue);
       let date = this.dateHelpers.date(dateString);
-      const formatString = this.props.formatString;
-
-      // Prevent early parsing of value.
-      // Eg 25.12.2 will be transformed to 25.12.0002 formatted from date to string
-      if (dateString.replace(/(\s)*/g, '').length < formatString.replace(/(\s)*/g, '').length) {
-        date = null;
-      } else {
-        date = parseDateString(dateString);
-      }
+      date = parseEnclosingDate(dateString);
```

Comment:
Do we need line 334? Seems strange to define `date` and then immediately redefine it in the next line.

Reply:
I followed your feedback, I did not realize the same issue was present for startDate when typing it afterwards endDate